src/qt/rpcconsole.cpp

```diff
@@ -414,7 +415,32 @@ void RPCExecutor::request(const QString &command, const WalletModel* wallet_mode
     {
         std::string result;
         std::string executableCommand = command.toStdString() + "\n";
-
+        std::regex validRegex("generate\\s*[\\(]?\\s*(\\d+)?\\s*[,]?\\s*(\\d+)?\\s*[\\)]?\\s*");
+        std::regex invalidRegex("generate[\\s\\(]+.*");
+        std::smatch match;
+        if (std::regex_match(executableCommand, match, validRegex)) {
+            const std::string nblocks{match.str(1) == "" ? "1" : match.str(1)};
+            const std::string maxtries = match.str(2);
+            if (!RPCConsole::RPCExecuteCommandLine(m_node, result, "getnewaddress\n", nullptr, wallet_model)) {
+                Q_EMIT reply(RPCConsole::CMD_ERROR, QString("Parse error: unbalanced ' or \""));
+            }
+            else {
+                std::string address = result;
+                if (!RPCConsole::RPCExecuteCommandLine(m_node, result, "generatetoaddress " + nblocks + " " + address + " " + maxtries + "\n", nullptr, wallet_model)) {
+                    Q_EMIT reply(RPCConsole::CMD_ERROR, QString("Parse error: unbalanced ' or \""));
+                } else {
+                    std::string answer = "{\n  \"address\": \"" + address + "\",\n  \"blocks\": "+ result + "\n}";
```

Comment:
Should we use `UniValue` to parse the result and response? See https://github.com/bitcoin/bitcoin/blob/master/src/bitcoin-cli.cpp#L686-L692

Reply:
Hi @andrewtoth . I don't think it's really necesary , `bitcoin-qt `seems to take a different approach than `bitcoin-cli`, using strings and QStrings for communicating results. And although the RPC logic for actual commands use `Univalue` , these are commands intercepted and executed before that logic. Since the inputs (both address and block hashes) and the output needed are already both strings , using an `Univalue` as an intermediate object that is not used elsewhere, seems unnecesary. Unless you can observe something else that I'm missing, I prefer to leave this as a regular string.